daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/VersionRange.scala

```diff
@@ -3,7 +3,36 @@
 
 package com.daml.lf
 
+import com.daml.lf.transaction.VersionTimeline
+
 final case class VersionRange[V](
     min: V,
     max: V,
-)
+) {
+  import VersionRange._
+  import Ordering.Implicits._
+
+  def intersect(that: VersionRange[V])(
+      implicit ev: VersionTimeline.SubVersion[V]): VersionRange[V] = {
+    VersionRange(
+      min = this.min max that.min,
```

Comment:
```suggestion
      min = maxVersion(this.min max that.min),
```

I didn't define a `minVersion` but you can do that if you want.